CMakeLists.txt

```diff
@@ -448,37 +448,47 @@ if(BUILD_DOCS)
         message(SEND_ERROR "Sphinx not found but BUILD_DOCS enabled")
     endif()
 
-    file(GLOB SPHINX_DEPS
-        "${CMAKE_CURRENT_SOURCE_DIR}/docs/*.rst"
-        "${CMAKE_CURRENT_SOURCE_DIR}/docs/guides/*.rst"
-        "${CMAKE_CURRENT_SOURCE_DIR}/docs/changelog.txt"
-        "${CMAKE_CURRENT_SOURCE_DIR}/docs/gen_changelog.py"
+    file(GLOB SPHINX_GLOB_DEPS
+        LIST_DIRECTORIES false
         "${CMAKE_CURRENT_SOURCE_DIR}/docs/images/*.png"
         "${CMAKE_CURRENT_SOURCE_DIR}/docs/styles/*"
-        "${CMAKE_CURRENT_SOURCE_DIR}/conf.py"
-        "${CMAKE_CURRENT_SOURCE_DIR}/scripts/about.txt"
-        "${CMAKE_CURRENT_SOURCE_DIR}/scripts/*/about.txt"
+        "${CMAKE_CURRENT_SOURCE_DIR}/data/init/*init"
+    )
+    file(GLOB_RECURSE SPHINX_GLOB_RECURSE_DEPS
+        "${CMAKE_CURRENT_SOURCE_DIR}/*.rst"
+        "${CMAKE_CURRENT_SOURCE_DIR}/changelog.txt"
+    )
+    list(FILTER SPHINX_GLOB_RECURSE_DEPS
+        EXCLUDE REGEX "docs/_"
     )
     file(GLOB_RECURSE SPHINX_SCRIPT_DEPS
         "${CMAKE_CURRENT_SOURCE_DIR}/scripts/*.lua"
         "${CMAKE_CURRENT_SOURCE_DIR}/scripts/*.rb"
+        "${CMAKE_CURRENT_SOURCE_DIR}/scripts/*.txt"
     )
-    set(SPHINX_DEPS ${SPHINX_DEPS} ${SPHINX_SCRIPT_DEPS}
-        "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE.rst"
+    set(SPHINX_DEPS ${SPHINX_GLOB_DEPS} ${SPHINX_GLOB_RECURSE_DEPS} ${SPHINX_SCRIPT_DEPS}
         "${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt"
+        "${CMAKE_CURRENT_SOURCE_DIR}/conf.py"
+        "${CMAKE_CURRENT_SOURCE_DIR}/docs/gen_changelog.py"
     )
 
     set(SPHINX_OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/docs/html/.buildinfo")
     set_source_files_properties(${SPHINX_OUTPUT} PROPERTIES GENERATED TRUE)
     add_custom_command(OUTPUT ${SPHINX_OUTPUT}
         COMMAND ${SPHINX_EXECUTABLE}
-            -a -E -q -b html
+            -E -q -b html -d "${CMAKE_BINARY_DIR}/docs/html"
```

Comment:
What was the reasoning for removing `-a`? f1cb9b9a839b2f80f2b4b5daab3000503b1353fc doesn't really explain it. My recollection is that `-a` was added to force output to be written to disk so that CMake recognizes when the target has been built and won't try to rebuild it on every build.